### PR TITLE
[doc] Fix the documentation about partition keys

### LINKED ISSUE #2237: [Feature] Improve the documentation about partition keys
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

image

Partition keys not have to be a subset of primary keys when primary keys are defined if the table is configured with dynamic bucket.

### Solution

Add corresponding description in `basic-concepts.md` and `creating-tables.md`. 

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by xiangyuf**
### Purpose

Linked issue: close #2237

Partition keys not have to be a subset of primary keys when primary keys are defined if the table is configured with dynamic bucket.

We need to fix this description about partition keys.

### CODE REVIEW

**Comment by xiangyuf** [comment]
Hi @JingsongLi, would u kindly help review this documentation update when u have time?

**Review by liming30 — APPROVED** [review]
+1

File: docs/content/concepts/basic-concepts.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ By partitioning, users can efficiently operate on a slice of records in the tabl
 
 {{< hint info >}}
 
-Partition keys must be a subset of primary keys if primary keys are defined.
+Partition keys must be a subset of primary keys if primary keys are defined and the table is using fixed bucket mode. For tables using dynamic bucket mode, partition keys only need to be different from the primary keys, see [Dynamic Bucket]({{< ref "concepts/primary-key-table#dynamic-bucket">}}).
```
Comment (JingsongLi) [inline docs/content/concepts/basic-concepts.md:43]: `If you want to use cross partition update, you should use dynamic bucket mode.`?

**Review by JingsongLi — COMMENTED** [review]
I think the causal relationship is a bit reversed. See my comment.

File: docs/content/concepts/basic-concepts.md (hunk 1 above)
Comment (xiangyuf) [reply docs/content/concepts/basic-concepts.md]: > `If you want to use cross partition update, you should use dynamic bucket mode.`?

@JingsongLi Hi, pls check the updated version.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**